backend/de.metas.business/src/main/java/de/metas/invoice/interceptor/C_Invoice.java

```diff
@@ -170,6 +171,25 @@ public void removeMaterialLinesNotCorrespondingToPriceList(final I_C_Invoice inv
 		}
 	}
 
+	@ModelChange(timings = { ModelValidator.TYPE_BEFORE_NEW })
+	public void setPaymentRule(final I_C_Invoice invoice)
```

Comment:
This bypasses invoice candidate once again.
Please don't set it via model interceptor but propagate it from the invoice candidate
(at backend/de.metas.swat/de.metas.swat.base/src/main/java/de/metas/invoicecandidate/api/impl/InvoiceCandBLCreateInvoices.java, from four first PR https://github.com/metasfresh/metasfresh/pull/12500#pullrequestreview-878008428 )